tools/azure-pipelines/jobs-template.yml

```diff
@@ -142,7 +142,7 @@ jobs:
       source ./tools/azure-pipelines/debug_files_utils.sh
       prepare_debug_files $(module)
       
-      ${{parameters.environment}} ./tools/ci/test_controller.sh $(module) ; TEST_EXIT_CODE=$?
+      ${{parameters.environment}} ./tools/ci/test_controller.sh $(module) $(Build.Reason); TEST_EXIT_CODE=$?
```

Comment:
I don't think you need to pass the variable into the script. The variable should be available as `$BUILD_REASON` in the scripts by default.